File: app/src/main/java/com/example/odm/garbagesorthelper/ui/search/SearchViewModelFactory.java
package com.example.odm.garbagesorthelper.ui.search;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.odm.garbagesorthelper.model.SearchDataRepository;

/**
 * description: 搜索页面ViewModel工厂类
 * author: ODM
 * date: 2019/10/20
 */
public class SearchViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private SearchDataRepository repository;

    public SearchViewModelFactory(SearchDataRepository repository) {
        this.repository = repository;
    }

    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        T viewModel;
        viewModel = (T)new SearchViewModel(repository);
        return viewModel;
    }
}

File: app/src/main/java/com/example/odm/garbagesorthelper/ui/search/SearchViewModelFactory.kt
package com.example.odm.garbagesorthelper.ui.search

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.example.odm.garbagesorthelper.base.BaseRepository
import com.example.odm.garbagesorthelper.model.SearchDataRepository

/**
 * description: 搜索页面ViewModel工厂类
 * author: ODM
 * date: 2019/10/20
 */
class SearchViewModelFactory(private val repository: SearchDataRepository) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        val viewModel: T

        viewModel = SearchViewModel(repository) as T
        return viewModel
    }

}